File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int,int>mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]=1;
        }
        int ans=0;
        for(int i=0;i<nums.size();i++){
            if(mp.find(nums[i]-1)!=mp.end()){
                continue;
            }
            else{
                int c=1;
                int ce=nums[i]+1;
                while(mp.find(ce)!=mp.end()){
                    c++;
                    ce++;
                }
                ans=max(ans,c);
            }
        }
        return ans;
        
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        if n == 0:
            return 0
        nums.sort()
        count = 1
        max_length = 1
        for i in range(1,n):
             if nums[i] != nums[i-1]:
                if nums[i] == nums[i-1]+1:
                    count +=1
                else: 
                    max_length = max(max_length,count)
                    count = 1
        return max(max_length,count)
                
        